File: spring-webflux/src/main/java/software/darkmatter/school/blog/domain/comment/data/Comment.java
package software.darkmatter.school.blog.domain.comment.data;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.relational.core.mapping.Table;
import software.darkmatter.school.blog.domain.user.data.User;

import java.time.OffsetDateTime;

@Data
@Table(name = "comments")
public class Comment {

    @Id
    private Long id;

    private Long postId;

    private String text;

    private OffsetDateTime createdAt;

    private Long createdByUserId;

    @Transient
    private User createdBy;

    private OffsetDateTime updatedAt;

    private Long updatedByUserId;

    private OffsetDateTime deletedAt;

    private Long deletedByUserId;
}

File: kotlin-spring-webflux/src/main/kotlin/software/darkmatter/school/blog/domain/comment/data/Comment.kt
package software.darkmatter.school.blog.domain.comment.data

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Transient
import org.springframework.data.relational.core.mapping.Table
import software.darkmatter.school.blog.domain.user.domain.User
import java.time.OffsetDateTime

@Table(name = "comments")
data class Comment(
    @Id
    var id: Long? = null,
    val postId: Long,
    var text: String,
    val createdAt: OffsetDateTime,
    val createdByUserId: Long,
    var updatedAt: OffsetDateTime,
    var updatedByUserId: Long,
    var deletedAt: OffsetDateTime?,
    var deletedByUserId: Long?,
) {
    @Transient
    var createdBy: User? = null
}
